scripts/database/upgrade.sh

```diff
@@ -0,0 +1,142 @@
+fi
+
+
+if ! grep -q '^'${upgrade_from}'$' releases.txt 
+then
+    echo >&2 "${upgrade_from} is not a valid Dataverse release"
+    exit 1
+fi
+
+if ! grep -q '^'${upgrade_to}'$' releases.txt 
+then
+    echo >&2 "${upgrade_to} is not a valid Dataverse release"
+    exit 1
+fi
+
+command -v psql >/dev/null 2>&1 || { 
+    echo >&2 'Cannot locate psql (PostgresQL command line utility)!';
+    echo >&2 'Make sure it is in your $PATH.';
+    echo >&2 'And if you have multiple versions of PostgresQL installed on your system,'; 
+    echo >&2 'make sure the psql in your $PATH is the same version your Dataverse is using.';
+    echo >&2 'Aborting...'; 
+    exit 1; 
+}
+
+echo "Enter the password for your PostgresQL database:"
+echo "(hit RETURN if you can access the database without a password)"
+read PGPASSWORD; export PGPASSWORD
+
+echo 
+echo "OK, let's verify that the PostgresQL credentials you provided are valid:"
+
+if dv_count=`psql -w -d ${pg_database} -U ${pg_user} -h ${pg_host} -p ${pg_port} -t -c "SELECT COUNT(*) FROM dataverse"`
+then
+    echo ok
+else 
+    echo >&2
+    echo >&2 "Failed to connect to the PostgresQL database!"
+    echo >&2 "Please verify your access credentials, and try again."
+    exit 1
+fi
+
+echo 
+echo "This script will attempt to upgrade the database ${pg_database},"
+echo "that currently has "`/bin/echo -n ${dv_count}`" dataverses, from version ${upgrade_from}"
+echo "to version ${upgrade_to}."
+echo "Hit RETURN to continue; or CTRL-C to exit"
+
+read line
+
+upgrade_flag="off"
+
+cat releases.txt | while read version
+do
+
+    if [ $upgrade_flag = "on" ]
+    then
+	# database create script: 
+	# (this will create the new tables, that were not present in the previous versions)
+
+	if [ ! -f create/create_${version}.sql ]
+	then
+	    echo >&2 "Cannot locate the create database script create/create_${version}.sql !"
+	    exit 1
+	fi
+
+	echo "Attempting to run the script create/create_${version}.sql..."
+
+	if ! psql -w d ${pg_database} -U ${pg_user} -h ${pg_host} -p ${pg_port} -f create/create_${version}.sql >/dev/null 2>&1
+	then
+	    echo >&2 "Failed to run the create database script for version ${version}"
+	    exit 1
+	fi
+
+	echo "ok"
+
+	# database ugprade script (if present): 
+	# (this will modify the tables that WERE present in the previous versions, 
+	# that were changed in this version (for example, if columns were added and/or 
+	# removed, if the data type of a certain column has changed, etc.)
+
+	if [ -f upgrades/upgrade_v*_to_${version}.sql ]
+	then
+	    echo "Attempting to execute the upgrade script:" upgrades/upgrade_v*_to_${version}.sql
+
+	    if ! psql -w -d ${pg_database} -U ${pg_user} -h ${pg_host} -p ${pg_port} -f upgrades/upgrade_v*_to_${version}.sql >/dev/null 2>&1
+	    then
+		echo >&2 "Failed to run the upgrade database script for version ${version}"
+		exit 1
+	    fi
+
+	    echo "ok"
+	else
+	    echo "(there is no database upgrade script for version ${version}...)"
```

Comment:
I'm a little concerned that no one but @landreev will know how to create create_v4.10.sql or create_v5.0.sql or whatever. It would be great to have something in doc/sphinx-guides/source/developers/making-releases.rst describing the process for making new database upgrade scripts. I assume the process is manual right now.

Reply:
I brought this up before, on slack: 
(and I will add it to the "making releases" section in the dev guide)
There will also be a procedure for saving the auto-generated database create script, and adding it to the branch, for every release.
It'll be very straightforward; since it needs to happen when a release is finalized, it'll be one of the last steps on the release checklist. Then we take the <GLASSFISH_ROOT>/domains/domain1/generated/ejb/dataverse/dataverse_VDCNet-ejbPU_createDDL.jdbc file on a node where the release candidate is deployed, and save it as scripts/database/create_vN.M.K.sql in the github tree.
(or that has been my idea... if anyone has objections/different ideas - please share; here or in the issue 4980)

Reply:
@landreev sounds good. Thanks.